src/wx/wayland.cpp

```diff
@@ -1,11 +1,31 @@
+#define private public
+#include <wx/window.h>
+#include <wx/glcanvas.h>
 #ifdef __WXGTK__
 #include <gdk/gdk.h>
 #ifdef GDK_WINDOWING_WAYLAND
 #include <gdk/gdkwayland.h>
-        bool IsItWayland() { return GDK_IS_WAYLAND_DISPLAY(gdk_display_get_default()); }
-#else
-        bool IsItWayland() { return false; }
+
+bool IsWayland() { return GDK_IS_WAYLAND_DISPLAY(gdk_display_get_default()); }
+
+// Temporary hack to backport 800d6ed69b from wxWidgets until 3.2.2 is released.
+#if !wxCHECK_VERSION(3, 2, 2)
```

Comment:
The check should be:
```c++
defined(__WXGTK__) && wxCHECK_VERSION(3, 2, 0) && !wxCHECK_VERSION(3, 2, 2)
```

Though I would suggest doing this instead before any other include:
```c++
#include <wx/version.h> 

#if defined(__WXGTK__)

#if wxCHECK_VERSION(3, 2, 2)
#define WX_HAS_WAYLAND_OPENGL_SUPPORT 1
#define WX_HAS_BROKEN_WAYLAND_OPENGL 0
#elif wxCHECK_VERSION(3,2,0)
#define WX_HAS_WAYLAND_OPENGL_SUPPORT 1
#define WX_HAS_BROKEN_WAYLAND_OPENGL 1
#else  // wxCHECK_VERSION
#define WX_HAS_WAYLAND_OPENGL_SUPPORT 0
#define WX_HAS_BROKEN_WAYLAND_OPENGL 0
#endif  // wxCHECK_VERSION

#else  // defined(__WXGTK__)

#define WX_HAS_WAYLAND_OPENGL_SUPPORT 0
#define WX_HAS_BROKEN_WAYLAND_OPENGL 0

#endif  // defined(__WXGTK__)
```

Then you can simply refer to the macros in the rest of the file, including for the includes.